contracts/exchangeIssuance/DEXAdapter.sol

```diff
@@ -420,27 +420,27 @@ library DEXAdapter {
         require(_path.length == 2, "ExchangeIssuance: CURVE_WRONG_PATH_LENGTH");
         (int128 i, int128 j) = _getCoinIndices(_pool, _path[0], _path[1], ICurveAddressProvider(_addresses.curveAddressProvider));
 
-        uint256 amountIn = _getAmountInCurve(
-            _pool,
-            i,
-            j,
-            _amountOut,
-            _addresses
-        );
-        require(amountIn <= _maxAmountIn, "ExchangeIssuance: CURVE_OVERSPENT");
 
         if(_path[0] == ETH_ADDRESS){
-            IWETH(_addresses.weth).withdraw(amountIn);
+            IWETH(_addresses.weth).withdraw(_maxAmountIn);
         }
 
-        uint256 returnedAmountOut = _exchangeCurve(i, j, _pool, amountIn, _amountOut, _path[0]);
+        uint256 returnedAmountOut = _exchangeCurve(i, j, _pool, _maxAmountIn, _amountOut, _path[0]);
         require(_amountOut <= returnedAmountOut, "ExchangeIssuance: CURVE_UNDERBOUGHT");
 
+        uint256 swappedBackAmountIn;
+        if(returnedAmountOut > _amountOut){
+            swappedBackAmountIn = _exchangeCurve(j, i, _pool, returnedAmountOut.sub(_amountOut), 0, _path[1]);
```

Comment:
Can we get wrecked here because we have minimumOutput = 0 ? 
I expect not since we are doing a reverse swap of a bigger swap we did before so sandwiching / frontrunning should not be possible in my limited understanding, but we should think this through.

Reply:
I thought through it as well and I agree with your initial instinct. That is to say that there is no way for a transaction to run between the two swaps and that no additional risk occurs by having the minimum out amount be 0. 

Alongside the introduction of a future feature that implements a minimum threshold to send dust back to the user, we could perhaps come up with a more reasonable minimum out from the pool.